consensus/src/experimental/commit_phase.rs

```diff
@@ -0,0 +1,333 @@
+                        }
+                        _ => {}
+                    };
+                }
+            }
+            let mut_local_queue = &mut self.local_queue;
+            let mut_local_cache = &mut self.local_cache;
+            loop {
+                let front = mut_local_queue.front().unwrap();
+                let front_chash = front
+                    .ledger_info_sig
+                    .ledger_info()
+                    .commit_info()
+                    .executed_state_id();
+                match mut_local_cache.entry(front_chash) {
+                    Entry::Occupied(lio) => {
+                        // cancel out an item from local_cache and an item from local_queue
+                        // TODO: given that every accumulated signatures has passed verification,
+                        // we might just need to count the number of signatures.
+                        if lio.get().verify_signatures(&self.verifier).is_ok() {
+                            commit(
+                                &self.execution_proxy,
+                                &front.vecblocks,
+                                &front.ledger_info_sig,
+                            )
+                            .await
+                            .map_err(|err| {
+                                counters::ERROR_COUNT.inc();
+                                error!(error = err.to_string(), "Error in committing blocks.")
+                            })
+                            .ok();
+                            assert!(
+                                self.committed_round
+                                    < front.ledger_info_sig.ledger_info().commit_info().round()
+                            );
+                            self.committed_round =
+                                front.ledger_info_sig.ledger_info().commit_info().round();
+                            self.commit_msg_sender
+                                .send(CommitPhaseChannelMsgWrapper::CommitDecision(
+                                    lio.get().clone(),
+                                ))
+                                .await
+                                .map_err(|err| {
+                                    counters::ERROR_COUNT.inc();
+                                    error!(
+                                        epoch = front
+                                            .ledger_info_sig
+                                            .ledger_info()
+                                            .commit_info()
+                                            .epoch(),
+                                        error = err.to_string(),
+                                        "Error occurred in sending the commit decision message."
+                                    )
+                                })
+                                .ok();
+                            lio.remove_entry();
+                            mut_local_queue.pop_front();
```

Comment:
this function is too long, maybe split into smaller functions

Reply:
resolved